### ISSUE TITLE
Nugetizer project not importing NuGet.Build.Packaging.targets? (CreateManifestResourceNames missing)

### ISSUE THREAD

**Issue by Andne** [NuGet/Home#5383]
## Details about Problem

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe): Nugetizer-3000

NuGet version (x.x.x.xxx): 0.1.313

VS version (if appropriate): 2017 15.3 Preview 2.0

OS version (i.e. win10 v1607 (14393.321)): Windows 7 SP1

Worked before? If so, with which NuGet version: No

## Detailed repro steps so we can see the same problem

1. Create new Nuget package project
2. Try to build, get error about missing CreateManifestResourceNames target
3. Edit .nuproj file, there are imports for Nuget.Packaging.Authoring.{props,targets}, but none for Nuget.Build.Packaging.{props,targets} even though it's installed in the project.  I have tried reinstalling the package as well with no effect.

I'm guessing that the Nuget.Build.Packaging.{props,targets} files are supposed to be imported in the project file so that the correct build scripts are used.  I was able to install the package to a .csproj file and see the imports show up, but for some reason the imports don't appear in a .nuproj.  Is this a bug and they're supposed to be there?  I do see NuGet.Packaging.Authoring.{props,targets} imported from a path that apparently is under AppData, but that seems to be for design-time only based on the contents and the condition on the import.

I originally saw this with VS2017 15.2, but this appears to be similar/same as to #5071 which implied that VS2017 15.3 Preview was a requirement for authoring new package projects.  I installed the preview and that has not helped.  I can probably edit the file and get the result I need, but it seems like this should work out of the box.

**Comment by Andne** [comment]
After manually adding the imports I started getting errors like this:
`Could not resolve mscorlib for target framework '.NETFramework,Version=v4.0'.`

The first occurred on saving the file after editing it, the second was when I tried to add references to a couple of projects in my solution.  The references did show up in the project file, but do not appear under the dependencies node.  Building the package project does build the library project and properly include the output in the package.

I've attached the VS error log from the events.
[VsProjectFault_de396307-d62b-4fa4-8cf8-0817557d88b9.failure.txt](https://github.com/NuGet/Home/files/1065253/VsProjectFault_de396307-d62b-4fa4-8cf8-0817557d88b9.failure.txt)

**Comment by kzu** [comment]
Hi @Andne, actually the `.nuproj` project uses a `PackageReference` to reference NuGetizer, so you will not see any imports in the `.nuproj` itself, but rather in the `obj\[projectname].nuget.g.props` and `obj\[projectname].nuget.g.targets` after a restore happened.

If you have a `packages.config` in the `.nuproj` instead of a `<PackageReference ...`, then the project won't work and you need to manually apply the fix mentioned in https://github.com/NuGet/Home/issues/5071#issuecomment-306864287 or just unfold the project again.

Please do let me know either way.

Thanks!

**Comment by Andne** [comment]
I've updated the version of NuGetizer in my 15.3 Preview 2 install of Visual Studio 2017 to 0.1.315, upon creating a new project it has a `packages.config` contained in it.  When I installed the preview, I did limit the features a little bit, is there a specific workload/component I need to make sure I have in order for the project to work correctly?  I see this behavior even when I create a brand new solution and nuget package project.

Here's the content of my project file, immediately after creation, with some values removed/replaced (just property values).  As best I can tell, it matches the current template in the repository that you referenced in the above link.
```
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
  <Import Project="$(NuGetAuthoringPath)\NuGet.Packaging.Authoring.props" Condition="Exists('$(NuGetAuthoringPath)\NuGet.Packaging.Authoring.props')" />
  <PropertyGroup Label="Globals">
    <ProjectGuid>2e18a258-ffa5-45fa-8533-faa64cd03320</ProjectGuid>
  </PropertyGroup>
  <PropertyGroup>
    <PackageId>(ProjectName)</PackageId>
    <PackageVersion Condition="'$(PackageVersion)' == ''">1.0.0</PackageVersion>
    <Title>(ProjectName)</Title>
    <Authors>Andne</Authors>
    <Owners>Andne</Owners>
    <Summary>(ProjectName)</Summary>
    <Description>(ProjectName</Description>
    <PackageReleaseNotes>
    </PackageReleaseNotes>
    <PackageProjectUrl>
    </PackageProjectUrl>
    <PackageLicenseUrl>
    </PackageLicenseUrl>
    <Copyright>Copyright © Andne</Copyright>
    <PackageTags>(ProjectName)</PackageTags>
    <RestoreProjectStyle>PackageReference</RestoreProjectStyle>
    <TargetFramework>netstandard1.0</TargetFramework>
    <PackageTargetFallback>net11;net20;net35;net40;net403;net45;net451;net452;net46;net461;net462;net47;netcore;netcore45;netcore451;netcore50;win8;win81;win10;sl4;sl5;wp;wp7;wp75;wp8;wp81;wpa81;uap;uap10;netstandard1.0;netstandard1.1;netstandard1.2;netstandard1.3;netstandard1.4;netstandard1.5;netstandard1.6;netstandard2.0;netcoreapp1.0;netcoreapp2.0;monoandroid;monotouch;monomac;xamarinios;xamarinmac;xamarinpsthree;xamarinpsfour;xamarinpsvita;xamarinwatchos;xamarintvos;xamarinxboxthreesixty;xamarinxboxone</PackageTargetFallback>
    <Configuration Condition="'$(Configuration)' == ''">Debug</Configuration>
    <OutputPath>bin\$(Configuration)\</OutputPath>
  </PropertyGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
  <ItemGroup>
    <PackageFile Include="Readme.txt" />
  </ItemGroup>
  <ItemGroup>
    <None Include="packages.config" />
  </ItemGroup>
  <Import Project="$(MSBuildToolsPath)\Microsoft.Common.targets" />
  <Import Project="$(NuGetAuthoringPath)\NuGet.Packaging.Authoring.targets" Condition="Exists('$(NuGetAuthoringPath)\NuGet.Packaging.Authoring.targets')" />
</Project>
```

**Comment by kzu** [comment]
I got a repro. Working on a fix

**Comment by kzu** [comment]
Ok, @Andne there's definitely something not working as expected in the 15.3 VSIX installer with regards to pre-requisites. Also, I'm not getting even the nested Clide.vsix installed :\

**Comment by Andne** [comment]
If that's the case, is there a work-around I can do to get all of the right components installed until they sort that out?

**Comment by kzu** [comment]
Here's a PR build in the works: https://ci.appveyor.com/project/MobileEssentials/nuget-build-packaging. Please try the resulting VSIX artifact once it finishes. That should fix the missing nested VSIX (Clide) but it would not fix the missing Merq pre-requisite.  You should get that dependency by issuing the following command:

```
vs_Enterprise.exe --add Microsoft.VisualStudio.Xamarin.Merq --installPath "C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise" --passive --wait
```

(see https://docs.microsoft.com/en-us/visualstudio/install/command-line-parameter-examples for more details on what that is doing and where to get vs_Enterprise.exe). You will need to change the .exe to the one used to install VS2017 preview in the first place.

**Comment by Andne** [comment]
After some bouncing around, I found that I need to install the Xamarin component from within the Visual Studio Installer.  This is maybe a heavy-weight method of getting Merq, but at least it should work.  The command-line you had didn't seem to do anything and when I extracted Merq.vsix and tried to install it directly, that's when I got the message about needing the Xamarin components installed.

Hopefully that can be cleaned up, it seems a bit wrong to need to install a 3GB component just to get one shared library.

Anyways, after installing the PR .vsix and Xamarin (I couldn't get the command-line for just the one component to work) I am able to create a project and it doesn't generate a packages.config file anymore.  I can add references and they are built into the package properly.  However, I don't see the project references in the solution explorer even though they show up in the references dialog.

**Comment by kzu** [comment]
Ok, we escalated the issue since that should be a fully supported scenario. We'll work it out hopefully before 15.3 ships to stable. Will keep you posted.

For now, yes, you'll need to install Xamarin :(

**Comment by kzu** [comment]
Ok, this has been confirmed to be an issue in current 15.3 preview. It will be fixed in Preview 4.

Will keep it open until we can resolve it fixed when that build is public.

Thanks for reporting @Andne !

**Comment by Andne** [comment]
I see that 15.3 is released (today apparently?), is this supposed to be fixed in that?  I've updated my main install and have tried to install the latest plugin but am still getting the same errors.  I've tried to uninstall and reinstall Nugetizer but it still lists that Microsoft.VisualStudio.Xamarin.Merq as needed.

I tried using the previously-referenced command line to add Merq manually, but that doesn't appear to work either, I get this error in dd_client_*date*.log `Microsoft.VisualStudio.Xamarin.Merq: No matching workload or component found.`.

Here's the log with the no matching component error in it:
[dd_client_20170815110539.txt](https://github.com/NuGet/Home/files/1225628/dd_client_20170815110539.txt)

I also found this web page: https://docs.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-professional and it does not appear to contain the component id `Microsoft.VisualStudio.Xamarin.Merq` or anything similar to it.
